src/Nest/Aggregations/AggregationsHelper.cs

```diff
@@ -177,16 +158,43 @@ private TAggregate TryGet<TAggregate>(string key)
 			return this.Aggregations.TryGetValue(key, out agg) ? agg as TAggregate : null;
 		}
 
-		private MultiBucketAggregate<TBucket> GetBucket<TBucket>(string key)
+		private MultiBucketAggregate<TBucket> GetMultiBucketAggregate<TBucket>(string key)
 			where TBucket : IBucket
 		{
 			var bucket = this.TryGet<BucketAggregate>(key);
 			if (bucket == null) return null;
 			return new MultiBucketAggregate<TBucket>
 			{
 				Buckets = bucket.Items.OfType<TBucket>().ToList(),
-				Meta = bucket.Meta
+				Meta = bucket.Meta,
 			};
+		}
+		private MultiBucketAggregate<KeyedBucket<TKey>> GetMultiKeyedBucketAggregate<TKey>(string key)
+		{
+			var bucket = this.TryGet<BucketAggregate>(key);
+			if (bucket == null) return null;
+			return new MultiBucketAggregate<KeyedBucket<TKey>>
+			{
+				Buckets = GetKeyedBuckets<TKey>(bucket.Items).ToList(),
+				Meta = bucket.Meta,
+			};
+		}
+
+
+		private IEnumerable<KeyedBucket<TKey>> GetKeyedBuckets<TKey>(IEnumerable<IBucket> items)
+		{
+			var buckets = items.OfType<KeyedBucket<object>>().ToList();
```

Comment:
I think this can be `.Cast<KeyedBucket<object>>()` here as we know it will be `KeyedBucket<object>`; this way, we skip the type check and just cast directly